homeassistant/components/sensor/noaa_tides.py

```diff
@@ -0,0 +1,128 @@
+"""
+This component provides HA sensor support for the NOAA Tides and Currents API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.noaa_tides/
+"""
+import logging
+from datetime import datetime, timedelta
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (ATTR_ATTRIBUTION, CONF_NAME, STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['py_noaa==0.2.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+API_URL = 'https://tidesandcurrents.noaa.gov/api/datagetter'
+CONF_STA_ID = 'station_id'
+CONF_TZ = 'timezone'
+CONF_UNITS = 'units'
+DEFAULT_ATTRIBUTION = "Data provided by NOAA"
+DEFAULT_NAME = 'NOAA Tides'
+DEFAULT_TZ = 'lst_ldt'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=60)
+
+TZS = ['gmt', 'lst', 'lst_ldt']
+UNITS = ['english', 'metric']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STA_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TZ, default=DEFAULT_TZ): vol.In(TZS),
+    vol.Optional(CONF_UNITS): vol.In(UNITS),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the NOAATidesAndCurrents sensor."""
+    sta = config[CONF_STA_ID]
+    name = config.get(CONF_NAME)
+    t_z = config.get(CONF_TZ)
+
+    if CONF_UNITS in config:
+        units = config.get(CONF_UNITS)
+    elif hass.config.units.is_metric:
+        units = UNITS[1]
+    else:
+        units = UNITS[0]
+
+    add_devices([NOAATidesAndCurrentsSensor(name, sta, t_z, units)], True)
+
+
+class NOAATidesAndCurrentsSensor(Entity):
+    """Representation of a NOAATidesAndCurrents sensor."""
+
+    def __init__(self, name, sta, t_z, units):
+        """Initialize the sensor."""
+        self._name = name
+        self._sta = sta
+        self._tz = t_z
+        self._units = units
+        self.data = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of this device."""
+        attr = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        if self.data['hi_lo'][1] == 'H':
```

Comment:
This will still be evaluated when state is unknown and `self.data` is `None`. So we will get an error here.